File: 1-easy/problems/0405-convert-a-number-to-hexadecimal/solution.ts
function toHex(num: number): string {
    if (num === 0) return "0";

    const hexChars = "0123456789abcdef";
    let result = "";

    const n = num >>> 0;
    let temp = n;

    while (temp) {
        result = hexChars[temp % 16] + result;
        temp = Math.floor(temp / 16);
    }

    return result;
}

File: 1-easy/problems/0405-convert-a-number-to-hexadecimal/solution.cpp
class Solution {
public:
    string toHex(int num) {
        if (num == 0) return "0";

        string hex_chars = "0123456789abcdef";
        string result = "";

        unsigned int n = (unsigned int)num;

        while (n) {
            result = hex_chars[n % 16] + result;
            n /= 16;
        }

        return result;
    }
};
